packages/api/src/tx/modules/ecocredit/v1/aminomessages.ts

```diff
@@ -0,0 +1,130 @@
+import { AminoConverters } from '@cosmjs/stargate';
+import { AminoMsg, Coin as AminoCoin } from '@cosmjs/amino';
+
+import {
+  MsgCreateClass,
+  MsgSend,
+  MsgSend_SendCredits,
+} from '../../../../generated/regen/ecocredit/v1/tx';
+import { Coin } from '../../../../generated/cosmos/base/v1beta1/coin';
+
+// Ref: https://github.com/regen-network/regen-ledger/blob/v4.0.0-beta1/x/ecocredit/core/codec.go#L16
+const msgCreateClassAminoType = 'regen.core/MsgCreateClass';
+const msgSendAminoType = 'regen.core/MsgSend';
+
+export interface AminoMsgCreateClass extends AminoMsg {
+  readonly type: typeof msgCreateClassAminoType;
+  readonly value: {
+    readonly admin: string;
+    readonly issuers: string[];
+    readonly metadata: string;
+    readonly credit_type_abbrev: string;
+    readonly fee?: AminoCoin;
+  };
+}
+
+interface AminoMsgSend_SendCredits {
+  $type: 'regen.ecocredit.v1.MsgSend.SendCredits';
+  batch_denom: string;
+  tradable_amount: string;
+  retired_amount: string;
+  retirement_jurisdiction: string;
+}
+
+export interface AminoMsgSend extends AminoMsg {
+  readonly type: typeof msgSendAminoType;
+  readonly value: {
+    readonly sender: string;
+    readonly recipient: string;
+    readonly credits: AminoMsgSend_SendCredits[];
+  };
+}
+
+export function isAminoMsgCreateClass(
+  msg: AminoMsg,
+): msg is AminoMsgCreateClass {
+  return msg.type === msgCreateClassAminoType;
+}
+
+export function isAminoMsgSend(msg: AminoMsg): msg is AminoMsgSend {
+  return msg.type === msgSendAminoType;
+}
+
+export function createEcocreditAminoConverters(): AminoConverters {
+  return {
+    '/regen.ecocredit.v1.MsgCreateClass': {
+      aminoType: msgCreateClassAminoType,
+      toAmino: ({
+        admin,
+        issuers,
+        metadata,
+        creditTypeAbbrev,
+        fee,
+      }: MsgCreateClass): AminoMsgCreateClass['value'] => {
+        return {
+          admin,
+          issuers,
+          metadata,
+          credit_type_abbrev: creditTypeAbbrev,
+          fee: fee && { amount: fee.amount, denom: fee.denom },
+        };
+      },
+      fromAmino: ({
+        admin,
+        issuers,
+        metadata,
+        credit_type_abbrev,
+        fee,
+      }: AminoMsgCreateClass['value']): Partial<MsgCreateClass> => {
+        return {
+          admin,
+          issuers,
+          metadata,
+          creditTypeAbbrev: credit_type_abbrev,
+          fee: fee && Coin.fromPartial(fee),
+        };
+      },
+    },
+    '/regen.ecocredit.v1.MsgSend': {
```

Comment:
@amaurym Would you be able to assist on this? I have a successful E2E test in amino.spec for `'/regen.ecocredit.v1.MsgSend'` . but when integrating with `regen-web`, I am seeing some errors. Something like "invalid characters."

Could you let me know if this is on the right track before I add more message types? Alternatively, if you have bandwidth, could you help us work on this issue? #35

Reply:
Sure! I can help. Could you provide the tx you sent, as well as the full error message?